### ISSUE TITLE
Write once run anywhere?

### ISSUE THREAD

**Issue by EasyG0ing1** [MrMarnic/JIconExtractReloaded#1]
This is Java .... you only support Windows?

**Comment by MrMarnic** [comment]
You are right. This is Java. But the underlying api for the icons is completly different on each operating system. I sadly don't know how to make it work on other operating systems than Windows.

**Comment by EasyG0ing1** [comment]
Try this ... I just tested it in Windows and MacOS and it works like a champ:

```
	public static Image getIcon(String filePath) {
		File                           file = new File(filePath);
		final javax.swing.JFileChooser fc   = new javax.swing.JFileChooser();
		javax.swing.Icon               icon = fc.getUI().getFileView(fc).getIcon(file);
		int width = icon.getIconWidth();
		int height = icon.getIconHeight();
		BufferedImage bufferedImage = new BufferedImage(
				width,
				height,
				BufferedImage.TYPE_INT_ARGB
		);
		icon.paintIcon(null, bufferedImage.getGraphics(), 0, 0);
		return SwingFXUtils.toFXImage(bufferedImage, null);
	}
```

That gives back a JavaFX Image ... if you still want the buffered image response, then use this:

```
	public BufferedImage getIconBuffered(String filePath) {
		File                           file = new File(filePath);
		final javax.swing.JFileChooser fc   = new javax.swing.JFileChooser();
		javax.swing.Icon               icon = fc.getUI().getFileView(fc).getIcon(file);
		int width = icon.getIconWidth();
		int height = icon.getIconHeight();
		BufferedImage bufferedImage = new BufferedImage(
				width,
				height,
				BufferedImage.TYPE_INT_ARGB
		);
		return bufferedImage;
	}
```

**Comment by EasyG0ing1** [comment]
@MrMarnic 

You could also keep your library the way it is since it lets you chose the icon size, and then test if the OS is Windows or not and if not, you could kick back a message saying something like This method only works in Windows, for other Operating Systems use ... whatever you wanna call the other method.

**Comment by ghost** [comment]
@MrMarnic it's been almost 2 years

**Comment by KnIfER** [comment]
for other system you can try the standard `Filesystemview`. 

[This file manger demo](https://github.com/konvio/javafx-file-manager) contains compiled .class files. you can test it using`java.exe  -classpath "out\production\DolphinExplorer;lib/*" com.konv.dolphinexplorer.Main`

 Filesystemview is buggy and returns cropped image on my computer 

<img width="297" height="146" alt="Image" src="https://github.com/user-attachments/assets/d6922826-640d-49db-b3ac-b37353fa27f6" />

(That's why I find this github via [google](https://www.google.com.hk/search?q=FileSystemView+getSystemIcon+cropped) => page 3 =>  [StackOverflow](https://stackoverflow.com/questions/52806549) ) .
